core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Platform.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids.tool.profiling
+
+/**
+ * Represents a platform and its associated recommendations.
+ */
+class Platform {
+  /**
+   * Recommendations to be excluded from the list of recommendations.
+   * These have the highest priority.
+   */
+  val recommendationsToExclude: Seq[String] = Seq.empty
+  /**
+   * Recommendations to be included in the final list of recommendations.
+   * These properties should be specific to the platform and not general Spark properties.
+   * For example: "spark.databricks.optimizer.dynamicFilePruning" for the Databricks platform.
+   *
+   * Represented as a tuple of (propertyKey, propertyValue).
+   */
+  val recommendationsToInclude: Seq[(String, String)] = Seq.empty
+  /**
+   * Dynamically calculates the recommendation for a specific Spark property by invoking
+   * the appropriate function based on `sparkProperty`.
+   * TODO: Implement this function and integrate with existing code in AutoTuner
+   *
+   * @param sparkProperty The Spark property for which the recommendation is calculated.
+   * @param args Variable list of arguments passed to the calculation function for dynamic
+   *             processing.
+   * @return Optional string containing the recommendation, or `None` if unavailable.
+   */
+  def getRecommendation(sparkProperty: String, args: Any*): Option[String] = None
+
+  /**
+   * Checks if the `property` is valid:
+   * 1. It should not be in exclusion list
+   *   OR
+   * 2. It should be in the inclusion list
+   */
+  def isValidRecommendation(property: String): Boolean = {
+    !recommendationsToExclude.contains(property) ||
+      recommendationsToInclude.map(_._1).contains(property)
+  }
+
+  /**
+   * Checks if the `comment` is valid:
+   * 1. It should not have any property from the exclusion list
+   */
+  def isValidComment(comment: String): Boolean = {
+    recommendationsToExclude.forall(excluded => !comment.contains(excluded))
+  }
+}
+class DatabricksPlatform extends Platform {
+  override val recommendationsToExclude: Seq[String] = Seq(
+    "spark.executor.cores",
```

Comment:
I think it is fine for now but would be good to move these to a separate variable where it could be used by other platforms as well. I am assuming some of the recommendationstoExclude will be common between all the platforms.

Reply:
mmm, yeah I agree that some types might share the same include/exclude policies.
However, we won't be able to group recommendations together as we haven't started investigating the other platforms too.

I am fine if this goes as is for now.